File: C++/Minimum Cost To Make Two Strings Identical.cpp
class Solution {
  public:
  int solve(string &x,string &y,vector<vector<int>>&dp,int i,int j){
      if(i>=x.length() || j>=y.length())
      return 0;
      if(dp[i][j]!=-1)
      return dp[i][j];
      int ans=0;
      if(x[i]==y[j])
      ans=1+solve(x,y,dp,i+1,j+1);
      else{
          ans=max(solve(x,y,dp,i+1,j),solve(x,y,dp,i,j+1));
      }
      return dp[i][j]=ans;
  }
    int findMinCost(string x, string y, int costX, int costY) {
        int n=x.length();
        int m=y.length();
        vector<vector<int>>dp(n,vector<int>(m,-1));
        int res=solve(x,y,dp,0,0);
         int ans=(n-res)*costX+(m-res)*costY;
         return ans;
    }
};

File: JAVA/Minimum Cost To Make Two Strings Identical.java
class Solution {
    public int findMinCost(String x, String y, int costX, int costY) {
        int n = x.length(), m = y.length();
        int dp[][] = new int[n + 1][m + 1];
        for(int i = 1; i <= n; i++) {
            for(int j = 1; j <= m; j++) {
                if(x.charAt(i - 1) == y.charAt(j - 1))
                    dp[i][j] = 1 + dp[i - 1][j - 1];
                else
                    dp[i][j] = Math.max(dp[i - 1][j], dp[i][j - 1]);
            }
        }
        int lcs = dp[n][m];
        int xLen = (n - lcs) * costX;
        int yLen = (m - lcs) * costY;
        return xLen + yLen;
    }
}
